app/Http/Controllers/ThreadController.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\ThreadRequest;
+use App\Models\Channel;
+use App\Models\Thread;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+
+class ThreadController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        return Thread::all();
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $this->response('Thread Creation');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(ThreadRequest $request)
+    {
+        Thread::create($request->all());
+    }
+
```

Comment:
Change the request input data method from `all` to `validated` so we are sure of what we are sending through.